services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -315,29 +321,31 @@ const deleteIndexedBlocks = async job => {
 				await transactionsTable.deleteByPrimaryKey(forkedTransactionIDs, dbTrx);
 				Signals.get('deleteTransactions').dispatch({ data: forkedTransactions });
 
-				// TODO: Update events fetching logic
-				// const events = await getEventsByHeight(block.height);
-				// if (events.length) {
-				// // TODO: Verify and enable it once pos:validatorStaked schema is exposed from SDK
-				// // Calculate locked amount change and update in key_value_store table for affected tokens
-				// const tokenIDLockedAmountChangeMap = {};
-				// events.forEach(event => {
-				// 	const { data: eventData } = event;
-				// 	// Initialize map entry with BigInt
-				// 	if ([EVENT_NAME.LOCKED, EVENT_NAME.UNLOCKED].includes(event.name)
-				// 		&& !(eventData.tokenID in tokenIDLockedAmountChangeMap) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] = BigInt(0);
-				// 	}
-
-				// 	// Negate amount to reverse the affect
-				// 	if (event.name === EVENT_NAME.LOCKED) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] -= eventData.amount;
-				// 	} else if (event.name === EVENT_NAME.UNLOCKED) {
-				// 		tokenIDLockedAmountChangeMap[eventData.tokenID] += eventData.amount;
-				// 	}
-				// });
-				// await updateTotalLockedAmounts(tokenIDLockedAmountChangeMap, dbTrx);
-				// }
+				// Calculate locked amount change from events and update in key_value_store table
+				const events = await getEventsByHeight(block.height);
+				if (events.length) {
+					const tokenIDLockedAmountChangeMap = {};
+					events.forEach(event => {
+						const { data: eventData } = event;
+						// Initialize map entry with BigInt
+						if ([EVENT_NAME.LOCK, EVENT_NAME.UNLOCK].includes(event.name)
+						&& !(eventData.tokenID in tokenIDLockedAmountChangeMap)) {
+							tokenIDLockedAmountChangeMap[eventData.tokenID] = BigInt(0);
+						}
+
+						// Negate amount to reverse the effect
+						if (event.name === EVENT_NAME.LOCK) {
+							tokenIDLockedAmountChangeMap[eventData.tokenID] -= BigInt(eventData.amount);
+						} else if (event.name === EVENT_NAME.UNLOCK) {
+							tokenIDLockedAmountChangeMap[eventData.tokenID] += BigInt(eventData.amount);
+						}
+					});
+					await updateTotalLockedAmounts(tokenIDLockedAmountChangeMap, dbTrx);
+				}
+
+				// Invalidate cached events of this block
+				// This must be done after processing all event related calculations
```

Comment:
```suggestion
				// Invalidate cached events for the current block
				// Must be done after the events are processed
```

Reply:
I would prefer keeping the previous comment as both are similar.